pkgs/games/shipwright/default.nix

```
@@ -0,0 +1,223 @@
+
+stdenv.mkDerivation rec {
+  pname = "shipwright";
+  version = "5.1.3";
+
+  src = fetchFromGitHub {
+    owner = "harbourmasters";
+    repo = "shipwright";
+    rev = version;
+    hash = "sha256-nDjaFrlBYXgxBW4BvsEI1uJVvXHgNwsro44B2Td85bI=";
+    fetchSubmodules = true;
+  };
+
+  enableParallelBuilding = true;
+
+  # Copied from the building guide
+  cmakeFlags = [
+    "DEBUG=0"
+    "OPTFLAGS=-O2"
+  ];
+
+  # Linking fails without this
+  hardeningDisable = [ "all" ];
+
+  # Many warnings are being treated as errors
+  NIX_CFLAGS_COMPILE = [
+    "-Wno-error"
+  ];
+
+  nativeBuildInputs = [
+    python3
+    llvmPackages.bintools
+    copyDesktopItems
+    cmake
+    ninja
+    pkg-config
+    lsb-release
+    clang
+    imagemagick
+    makeWrapper
+  ];
+
+  buildInputs = [
+    libX11
+    libXrandr
+    libXinerama
+    libXcursor
+    libXi
+    libXext
+    glew
+    SDL2
+    SDL2_net
+    libpulseaudio
+    libpng
+    bzip2
+    StormLib
+  ];
+
+  postPatch = ''
+    # This script attempts to call git to retrieve the commit hash,
+    # so we have to manually patch it in instead.
+    substituteInPlace ZAPDTR/ZAPD/genbuildinfo.py \
+      --replace "label = subprocess" "#label = subprocess" \
+      --replace "+ label +" "+ \"${builtins.substring 0 7 src.rev}\" +"
+
+    # SHIPWRIGHT: Stop ship from looking at user dir for OTRs
+    substituteInPlace soh/soh/OTRGlobals.cpp \
+      --replace "std::string mqPath = Ship::Window::GetPathRelativeToAppDirectory(\"oot-mq.otr\");" "std::string mqPath = \"oot-mq.otr\";" \
+      --replace "std::string ootPath = Ship::Window::GetPathRelativeToAppDirectory(\"oot.otr\");" "std::string ootPath = \"oot.otr\";"
+
+    # Copy Rom(s) to OTRExporter for asset generation.
+    ln -s ${ootRom} OTRExporter/oot.z64
+    ${lib.optionalString enableDualMqRom "ln -s ${ootMqRom} OTRExporter/oot-mq.z64" }
+  '';
+
+  # Extract assets. This does compile a fair bit of code, which is why it is not done in the configure phase.
+  preBuild = ''
+    cmake --build . --target ExtractAssets -v ''${cmakeFlagsArray[@]}
+  '';
+
+  installPhase = ''
+    runHook preInstall
+    mkdir -p $out/{bin,share/pixmaps}
+    cd soh
+
+    # Copy icon for desktop entry
+    convert ../../soh/icon.jpg $out/share/pixmaps/soh.png
+
+    # Copy respective OTR(s)
+    ${lib.optionalString (!isMqVariant) "cp oot.otr $out/share"}
+    ${lib.optionalString (enableDualMqRom || isMqVariant ) "cp oot-mq.otr $out/share"}
+
+    # oot.otr needs to be in the same directory as the executable
+    cp soh.elf $out/share
+
+    # create wrapper to set SHIP_HOME env var
+    makeWrapper $out/share/soh.elf $out/bin/soh \
+      --chdir $out/share \
+      --argv0 soh
+
+    # Workaround because makeWrapper uses single quotes instead of double when using `--set`
+    # Needed because $HOME needs to be replaced with users home on runtime.
+    sed -i '1 a export SHIP_HOME="$HOME/.local/share/ShipOfHarkinian"' $out/bin/soh
```

Cleaner approach to the wrapper, in combination with the suggestion above. 
```suggestion
    cp soh.elf $out/bin/soh
    
    # Without SHIP_HOME set soh attempts to write to the nix store. This is an ad-hoc version of makeWrapper's --set-default flag, 
    # required because that does not allow for variable expansion at runtime.
    wrapProgram $out/bin/soh --run 'export SHIP_HOME="''${SHIP_HOME:-"$HOME/.local/share/ShipOfHarkinian"}"'
```